File: app/src/main/java/com/liarstudio/courierservice/logic/pack/PackageAPI.java
package com.liarstudio.courierservice.API;

import com.liarstudio.courierservice.entities.pack.Package;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface PackageAPI {


    //Получаем список всех посылок
    @GET("/package")
    Call<List<Package>> loadPackages();

    //Получаем одну посылку по id
    @GET("/package")
    Call<Package> loadPackages(@Query("pack_id") int id);

    //Получаем все посылки по id курьера и статусу(либо новая, либо нет)
    @GET("package/courier/{courier_id}")
    Call<List<Package>> loadCourierPackages(@Path("courier_id") int id, @Query("status") int status);

    //Получаем посылки для администратора(со статусом "Новая"/"Отклоненная"/"Завершенная"
    @GET("package/admin")
    Call<List<Package>> loadAdminPackages();

    //Удаляем посылку
    @DELETE("package/admin")
    Call<Void> delete(@Query("pack_id") int id);

    //Добавляем посылку
    @POST("/package")
    Call<Void> add(@Body Package pack);
}

File: app/src/main/java/com/liarstudio/courierservice/logic/pack/PackageApi.kt
package com.liarstudio.courierservice.logic.pack

import com.liarstudio.courierservice.logic.UrlPackage
import com.liarstudio.courierservice.logic.UrlPackage.GET_USER_PACKAGES
import com.liarstudio.courierservice.logic.pack.request.PackageRequest
import com.liarstudio.courierservice.logic.pack.response.PackResponse
import io.reactivex.Completable

import io.reactivex.Observable
import retrofit2.http.*

interface PackageApi {

    /**
     * Получение посылки по id
     *
     * @param id идентификатор посылки
     *
     * @return Observable с посылкой
     */
    @GET(UrlPackage.GET_PACKAGE)
    fun getPackage(@Path("package_id") id: Long): Observable<PackResponse>

    /**
     * Получение всех посылок пользователя
     *
     * @param id идентификатор пользователя
     * @param statuses статусы посылок, которые необходимо получить
     *
     * @return Observable со списком посылок
     */
    @GET(GET_USER_PACKAGES)
    fun getUserPackages(
            @Path("user_id") id: Long,
            @Query("status") vararg statuses: Int
    ): Observable<List<PackResponse>>

    /**
     * Удаление посылки по id
     *
     * @param id идентификатор посылки
     *
     * @return Completable
     */
    @DELETE(UrlPackage.BASE_PACKAGE_URL)
    fun delete(@Path("package_id") id: Long): Completable

    /**
     * Добавление или изменение посылки
     *
     * @param packageRequest маппинг-модель посылки
     *
     * @return Completable
     */
    @POST(UrlPackage.BASE_PACKAGE_URL)
    fun add(@Body packageRequest: PackageRequest): Completable
}
